### PR TITLE
Add support for ppc64le clobber_abi

### PR DESCRIPTION
**Pull request by ecnelises**

### CODE REVIEW

**Comment by ecnelises** [comment]
CC'ing @bzEq

**Comment by michaelwoerister** [comment]
r? compiler

File: compiler/rustc_target/src/asm/mod.rs [hunk 1]
```diff
@@ -1084,6 +1091,13 @@ impl InlineAsmClobberAbi {
                     f16, f17, f18, f19, f20, f21, f22, f23,
                 }
             },
+            InlineAsmClobberAbi::PowerPC64ElfV2 => clobbered_regs! {
+                PowerPC PowerPCInlineAsmReg {
+                    r0, r3, r4, r5, r6, r7, r8, r9, r10, r11, r12,
+                    xer, cr0, cr1, cr5, cr6, cr7,
+                    f0, f1, f2, f3, f4, f5, f6, f7, f8, f9, f10, f11, f12, f13,
+                }
```
Comment (lu-zero) [inline compiler/rustc_target/src/asm/mod.rs:1099]: What about the vector registers?

File: compiler/rustc_target/src/asm/mod.rs [hunk 2]
```diff
@@ -927,6 +928,12 @@ impl InlineAsmClobberAbi {
                 "C" | "system" => Ok(InlineAsmClobberAbi::LoongArch),
                 _ => Err(&["C", "system"]),
             },
+            InlineAsmArch::PowerPC64 => match name {
+                "C" | "system" if target.endian == Endian::Little => {
```
Comment (lu-zero) [inline compiler/rustc_target/src/asm/mod.rs:932]: The list as-is would be fine for both endiannesses.

**Comment by Dylan-DPC** [comment]
@ecnelises any updates on this? thanks

**Comment by ecnelises** [comment]
Closed in favor of #131341

**Pull request closed without merging**